2 dz/5/main.cpp

```diff
@@ -0,0 +1,54 @@
+/*Реализовать алгоритм пирамидальной сортировки.*/
+#include <iostream>
+
+using namespace std;
+
+void max_heap(int *arr, int n, int i)
+{
+    int largest = i;
+    int l = 2 * i + 1;
+    int r = 2 * i + 2;
+    if (l < n && arr[l] > arr[largest])
+    {
+        largest = l;
+    }
+    if (r < n && arr[r] > arr[largest])
+    {
+        largest = r;
+    }
+    if (largest != i)
+    {
+        swap(arr[i], arr[largest]);
+        max_heap(arr, n, largest);
+    }
+}
+
+void heapsort(int *row, int n)
```

Comment:
It's something different. You don't need to build the max heap every time, just one time.
See: https://neerc.ifmo.ru/wiki/index.php?title=%D0%A1%D0%BE%D1%80%D1%82%D0%B8%D1%80%D0%BE%D0%B2%D0%BA%D0%B0_%D0%BA%D1%83%D1%87%D0%B5%D0%B9#.D0.A0.D0.B5.D0.B0.D0.BB.D0.B8.D0.B7.D0.B0.D1.86.D0.B8.D1.8F
And maybe the animation at https://ru.wikipedia.org/wiki/%D0%9F%D0%B8%D1%80%D0%B0%D0%BC%D0%B8%D0%B4%D0%B0%D0%BB%D1%8C%D0%BD%D0%B0%D1%8F_%D1%81%D0%BE%D1%80%D1%82%D0%B8%D1%80%D0%BE%D0%B2%D0%BA%D0%B0